### PR TITLE
Generate head links for Continue Edit Session

### LINKED ISSUE #159685: Continue Working On from desktop to vscode.dev doesn't preserve branch checkout
1. Have a desktop clone open with some uncommitted changes
2. Run Continue Edit Session > open on vscode.dev
3. :bug: edit session is not applied 

The reason for this is that we are opening a permalink instead of preserving the current branch checkout so the edit session identifier doesn't match up and the edit session doesn't get applied. This used to work and regressed with the edit session identifier API work, which applies stricter checks to edit session identities. Thanks @lramos15 for finding this in https://github.com/microsoft/vscode/issues/159287!

### PR DESCRIPTION
**Pull request by joyceerhl**
This PR fixes https://github.com/microsoft/vscode/issues/159685

### CODE REVIEW

File: extensions/github/src/links.ts [hunk 1]
```diff
@@ -125,11 +125,11 @@ export function getPermalink(gitAPI: GitAPI, useSelection: boolean, hostPrefix?:
 		return;
 	}
 
-	const commitHash = (gitRepo.state.HEAD?.ahead === 0) ? `/blob/${gitRepo.state.HEAD?.commit}` : '';
+	const blobSegment = (gitRepo.state.HEAD?.ahead === 0) ? `/blob/${linkType === 'headlink' ? gitRepo.state.HEAD.name : gitRepo.state.HEAD?.commit}` : '';
```
Comment (DonJayamanne) [inline extensions/github/src/links.ts:128]: Don't we need to url encode the name? from what I know branch names can contain special characters.

File: extensions/github/src/commands.ts [hunk 2]
```diff
@@ -26,8 +26,8 @@ async function copyVscodeDevLink(gitAPI: GitAPI, useSelection: boolean) {
 
 async function openVscodeDevLink(gitAPI: GitAPI): Promise<vscode.Uri | undefined> {
 	try {
-		const permalink = getPermalink(gitAPI, true, getVscodeDevHost());
-		return permalink ? vscode.Uri.parse(permalink) : undefined;
+		const headlink = getLink(gitAPI, true, getVscodeDevHost(), 'headlink');
+		return headlink ? vscode.Uri.parse(headlink) : undefined;
```
Comment (ProKashif) [inline extensions/github/src/commands.ts:30]:
```suggestion
		return headlink && vscode.Uri.parse(headlink);
```

File: extensions/github/src/links.ts (hunk 1 above)
Comment (ProKashif) [reply extensions/github/src/links.ts:128]: +

**Review by ProKashif — CHANGES_REQUESTED** [review]

File: extensions/github/src/links.ts (hunk 1 above)
Comment (joyceerhl) [reply extensions/github/src/links.ts:128]: This doesn't seem required elsewhere, e.g. https://insiders.vscode.dev/github.com/microsoft/vscode-pull-request-github/blob/57dd87743f1caa943ef5e65bfc0d6dcf101a0ee6/src/issues/util.ts#L559-L576

Comment (DonJayamanne) [reply extensions/github/src/links.ts:128]: Sure, but thats an issue isn't it.

**Review by DonJayamanne — APPROVED** [review]
Looks like there might be an issue with branch names that can contain special characters that might not be url friendly.
However that seems to be an existing issue.

**Pull request merged**